x-pack/auditbeat/module/system/socket/socket.go

```diff
@@ -176,6 +178,18 @@ func (s Socket) toMapStr() common.MapStr {
 	return mapstr
 }
 
+// entityID creates an ID that uniquely identifies this socket across machines.
+func (s Socket) entityID(hostID string) string {
+	h := system.NewEntityHash()
+	binary.Write(h, binary.LittleEndian, []byte(hostID))
+	binary.Write(h, binary.LittleEndian, int64(s.Inode))
+	h.Write(s.LocalIP)
+	h.Write(s.RemoteIP)
+	binary.Write(h, binary.LittleEndian, int64(s.LocalPort))
+	binary.Write(h, binary.LittleEndian, int64(s.RemotePort))
```

Comment:
From what I can see the community ID is using source and dest IP/port, and protocol. For us, we would want to be host specific though, since even if we have both sockets of a connection in the data, we would still want them to have different entity IDs (since they are different entities, owned by different processes and users). That said, we should probably add community ID to the socket fields, to allow correlation between flows and sockets. I think @andrewkroh has suggested it in the past.

Reply:
Having the community ID populated wherever possible will make correlation so much easier. So +1 to adding it for the socket dataset.

Reply:
This is extra goodness, not a breaking change, though. So up to @cwurm to include now vs wait for 7.1? I think both are fine, it's just a question of time management. I think it's pretty late to add community_id, unless code reuse is more straightforward than I assume :-)

Reply:
Yeah, I'll put it in the backlog.